### PR TITLE
Stun Batons now only trigger when in harm intent

### LINKED ISSUE #5721: Attempting to use stun batons on people while they're off should only attack while on harm intent.
### Description
Title. In /tg/, you can only actually attack people with stun batons while they are turned off if you are on harm intent when clicking them. If the baton is turned off and you try using it on people on any other intent, then nothing actually happens except a feedback message appearing in chat.
image
The above messages are formatted as such
For anyone that is not the target: "[Attacker] prods [Target] with the [Baton Name]. Luckily it was off."
For anyone that is the target: "[Attacker] prods you with the [Baton Name]. Luckily it was off."

This should prevent novices from harming other people while trying to stun them with a turned-off baton while also informing them that yes, their baton is indeed turned off.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
closes #5721

CL: [Improvement] Stun Batons now only trigger when in harm intent.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs [hunk 1]
```diff
@@ -118,14 +118,22 @@ public void ServerPerformInteraction(HandApply interaction)
...
+			if (interaction.Intent == Intent.Harm && toggleableEffect.CurrentWeaponState == ToggleableEffect.WeaponState.On)
 			{
 				// Direction of attack towards the attack target.
 				wna.ServerPerformMeleeAttack(target, dir, interaction.TargetBodyPart, LayerType.None);
 			}
+			else
+			{
+				Chat.AddActionMsgToChat(interaction.Performer,
+					$"You attempt to prod {interaction.TargetObject.ExpensiveName()} but the {gameObject.ExpensiveName()} was off!",
+					$"{interaction.Performer.ExpensiveName()} prods {interaction.TargetObject.ExpensiveName()}, luckily the {gameObject.ExpensiveName()} was off!");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs:135]: This will trigger when not Harm but On and then the message doesnt make sense.

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs:135]: Ah good point, will fix it after I'm done looking at another issue

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs [hunk 2]
```diff
@@ -118,14 +118,22 @@ public void ServerPerformInteraction(HandApply interaction)
 			Vector2 dir = (target.transform.position - performer.transform.position).normalized;
 
 			WeaponNetworkActions wna = performer.GetComponent<WeaponNetworkActions>();
+			ToggleableEffect toggleableEffect = gameObject.GetComponent<ToggleableEffect>();
 
 			// If we're on harm intent we deal damage!
 			// Note: this has to be done before the teleport, otherwise the target may be moved out of range.
-			if (interaction.Intent == Intent.Harm)
+			if (interaction.Intent == Intent.Harm && toggleableEffect.CurrentWeaponState == ToggleableEffect.WeaponState.On)
```
Comment (AtnerNT) [inline UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs:125]: One problem, when toggleableEffect.CurrentWeaponState != ToggleableEffect.WeaponState.On, the meleeEffect component will actually be disabled.  So the feedback message won't show.

**Review by AtnerNT — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs (hunk 2 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs:125]: but it showed on my end?

Comment (AtnerNT) [reply UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs:125]: It showed when the baton was turned off?

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs:135]: Fixed

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs (hunk 2 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs:125]: oh i get what you mean now.
I didn't notice that the component gets disabled when the baton is off, will find a way to keep it working.

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs:125]: Fixed

**Review by Bod9001 — APPROVED** [review]

**Comment by ThatDan123** [comment]
@AtnerNT This look good now?

**Comment by AtnerNT** [comment]
> @AtnerNT This look good now?

Yeah nah everything seems good, tested it and it all seems to work fine.

**Review by ThatDan123 — APPROVED** [review]

**Pull request merged**